include/alpaka/idx/bt/IdxBtOmp5BuiltIn.hpp

```diff
@@ -0,0 +1,145 @@
+                /*virtual*/ ~IdxBtOmp5BuiltIn() = default;
+            };
+        }
+    }
+
+    namespace dim
+    {
+        namespace traits
+        {
+            //#############################################################################
+            //! The GPU CUDA accelerator index dimension get trait specialization.
+            template<
+                typename TDim,
+                typename TIdx>
+            struct DimType<
+                idx::bt::IdxBtOmp5BuiltIn<TDim, TIdx>>
+            {
+                using type = TDim;
+            };
+        }
+    }
+    namespace idx
+    {
+        namespace traits
+        {
+            //#############################################################################
+            //! The GPU CUDA accelerator block thread index get trait specialization.
+            template<
+                typename TDim,
+                typename TIdx>
+            struct GetIdx<
+                idx::bt::IdxBtOmp5BuiltIn<TDim, TIdx>,
+                origin::Block,
+                unit::Threads>
+            {
+                //-----------------------------------------------------------------------------
+                //! \return The index of the current thread in the block.
+                template<
+                    typename TWorkDiv>
+                static auto getIdx(
+                    idx::bt::IdxBtOmp5BuiltIn<TDim, TIdx> const &,
+                    TWorkDiv const & workDiv)
+                -> vec::Vec<TDim, TIdx>
+                {
+                    // We assume that the thread id is positive.
+                    ALPAKA_ASSERT(::omp_get_thread_num()>=0);
+                    // \TODO: Would it be faster to precompute the index and cache it inside an array?
+                    return idx::mapIdx<TDim::value>(
+                        vec::Vec<dim::DimInt<1u>, TIdx>(static_cast<TIdx>(::omp_get_thread_num())),
+                        workdiv::getWorkDiv<Block, Threads>(workDiv));
+                }
+            };
+
+            template<
+                typename TIdx>
+            struct GetIdx<
+                idx::bt::IdxBtOmp5BuiltIn<dim::DimInt<1u>, TIdx>,
+                origin::Block,
+                unit::Threads>
+            {
+                //-----------------------------------------------------------------------------
+                //! \return The index of the current thread in the block.
+                template<
+                    typename TWorkDiv>
+                static auto getIdx(
+                    idx::bt::IdxBtOmp5BuiltIn<dim::DimInt<1u>, TIdx> const & idx,
+                    TWorkDiv const &)
+                -> vec::Vec<dim::DimInt<1u>, TIdx>
+                {
+                    alpaka::ignore_unused(idx);
+                    return vec::Vec<dim::DimInt<1u>, TIdx>(static_cast<TIdx>(omp_get_thread_num()));
+                }
+            };
```

Comment:
Could you add this 1D specialization to `alpaka::idx::bt::IdxBtOmp` and get rif of `IdxBtOmp5BuiltIn` completely?
This could also be a separate PR.

Reply:
https://github.com/alpaka-group/alpaka/pull/1140